### ISSUE TITLE
`selector` only identifies the first matching dynamically generated element (Cypress)

### ISSUE THREAD

**Issue by dmwelch** [applitools/eyes.sdk.javascript1#250]
Hello, 

I'm trying to do something like https://applitools.com/wp-content/uploads/2020/05/image-8.png from [Annotations (continued)](https://applitools.com/automated-visual-testing-best-practices-guide/#Annotations) and I don't see any documentation on how to do this with the Cypress implementation.  Here's what I'm doing:

```javascript
describe('Dashboard visual tests', () => {
  afterEach(() => {
    cy.eyesClose()
  })

  it('Dashboard with many cards', () => {
    const layouts = [
      { selector: '.card-container' },  // Multiple, dynamically generated components
    ]

    const ignores = [
      { selector: '.card-label-name' },  // inside each .card-container is one of these
      { selector: '.label-container.data-label.data-label-value' },  // inside each .card-container is many of these
    ]
    
    cy.eyesOpen({
      appName: 'My App',
      testName: 'Home page',
      batchName: 'Home page',
    })

    // Check the app page
    cy.eyesCheckWindow({
      ignore: ignores,
      // layout: layouts,
      tag: 'Initial view',
      target: 'window',
      fully: 'false',
    })
  })
})
```

I thought I could do something like
```javascript
const ignores = [
      { byClass: '.card-label-name' },  // inside each .card-container is one of these
      { byClass: '.label-container.data-label.data-label-value' },  // inside each .card-container is many of these
    ]
```
but no dice.  Any help would be appreciated!

**Comment by amitzur** [comment]
@dmwelch thanks for letting us know. This is currently a limitation of the Cypress SDK - only one region is created per selector.
This is on our plans to fix in the near future.

**Comment by dmwelch** [comment]
Thanks!  I can't stress how critical this is to our testing strategy - is there a workaround for the time being or are we just out of luck?

**Comment by amitzur** [comment]
@dmwelch The workaround you could do as an immediate step, is to find all `.card-container` before calling `cy.eyesCheckWindow`, then mark all of them with some attribute which is unique for each element, and specify the array of all elements as the `layout` argument. The same goes for `ignore`.

Something along the lines of:

```
cy.get('.card-container').then(elements => {
  const ids = elements.map(element => {
    const id = `${Math.random()}`.substr(2)
    element.setAttribute('data-unique-selector', id)
    return id
  })

  cy.eyesCheckWindow({
    layout: ids.map(id => `[data-unique-selector="${id}"]`)
    // ... rest of the arguments
  })
})
```

**Comment by dmwelch** [comment]
Couldn't get the workaround to work.   Will post more when I return to it.

**Comment by amitzur** [comment]
@dmwelch We are about to release a new version shortly which will support this!
I'll update here.

**Comment by amitzur** [comment]
@dmwelch this is released in version `3.15.2`!
From now on, every element matching the selector provided will be assigned a region.
